docs/public_api.md

```diff
@@ -0,0 +1,139 @@
+Validator stores API key information in the SQL lite database stored on the validator node. File name: `api_keys.db`. And it's stored in the .bittensor folder. If you validator coldkey is named `validator`, default location would be here: ~/.bittensor/miners/validator/default/netuid17/validator/api_keys.db
+
+We provide a simple `api_key_cli.sh` located in `/three-gen-subnet/scripts` dir to work with API keys database.
+
+Validator synchronizes API keys with the database every 30 minutes.
+
+All commands require specifying the path to the database. 
+
+## Generate new API key
+```commandline
+./api_key_cli.sh ~/.bittensor/miners/validator/default/netuid17/validator/api_keys.db gen
+```
+
+## List all API keys
+```commandline
+./api_key_cli.sh ~/.bittensor/miners/validator/default/netuid17/validator/api_keys.db list
+```
+
+## Add new API key
+To add new client API key you need to provide some client name or alias, and specify rate limit parameters (number of requests per defined period).
+The first parameter defines the number of requests, the second interval in seconds.
+```commandline
+./api_key_cli.sh ~/.bittensor/miners/validator/default/netuid17/validator/api_keys.db add EWJzrqUL94r5BEz0ZLPUbEwD8qdQX18dCUF9B7C2e0 client-name 100 60
+```
+In this example, client would be allowed to do 100 requests per 60 seconds (1 minute).
+
+## Remove API key
+```commandline
+./api_key_cli.sh ~/.bittensor/miners/validator/default/netuid17/validator/api_keys.db remopve EWJzrqUL94r5BEz0ZLPUbEwD8qdQX18dCUF9B7C2e0
+```
+
+# Integrating with the public endpoint
+
+## How it works
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant Validator
+    participant Miner1
+    participant Miner2
+
+    Client->>Validator: Auth
+    Client->>Validator: PromptData
+    Miner1->>Validator: Pull task
+    Validator->>Miner1: Send task
+    Validator->>Client: TaskUpdate (STARTED)
+    Miner2->>Validator: Pull task
+    Validator->>Miner2: Send task
+    Miner1->>Validator: Submit results
+    alt If configured to do so
+        Validator->>Client: TaskUpdate (FIRST_RESULTS)
+    end
+    Miner2->>Validator: Submit results
+    Validator->>Validator: Select best results
+    Validator->>Client: TaskUpdate (BEST_RESULTS)
+```
+
+
+### Explanation of the Sequence:
+
+1. **Client Initiates Connection**:
+    - `Client` connects to `Validator`.
+    - `Client` sends authentication info (`Auth`).
+    - `Client` sends `PromptData` to `Validator`.
+
+2. **Miner 1 Interactions**:
+    - `Miner 1` pulls a task from `Validator`.
+    - The `Validator` sends the task details to `Miner 1`.
+    - The `Validator` sends a task update (`STARTED`) to the `Client`.
+
+3. **Miner 2 Interactions**:
+    - `Miner 2` pulls a task from `Validator`.
+    - The `Validator` sends the task details to `Miner 2`.
+
```

Comment:
missing:   - The `Validator` sends a task update (`STARTED`) to the `Client`
yo make both miner1 and miner2 conistent